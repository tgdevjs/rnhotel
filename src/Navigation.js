import React from 'react';
import {
  createAppContainer,
  createStackNavigator,
  createBottomTabNavigator,
} from 'react-navigation';
import Ionicons from 'react-native-vector-icons/MaterialIcons';

import {
  Hotels,
  ConfirmReservationQuery,
  SearchCalendar,
  SearchWithQuery,
  AccountWithQuery,
  SignInModalWithQuery,
  JoinModalWithQuery,
  StaysWithQuery,
} from './screens';

const SearchStack = createStackNavigator({
  Search: {
    screen: SearchWithQuery,
  },
  Calendar: {
    screen: SearchCalendar,
  },
  Hotels: {
    screen: Hotels,
  },
  ConfirmReservation: {
    screen: ConfirmReservationQuery,
  },
});

const AccountStack = createStackNavigator({
  Account: {
    screen: AccountWithQuery,
  },
});

const StaysStack = createStackNavigator({
  Stays: {
    screen: StaysWithQuery,
  },
});

const icons = { Search: 'search', Account: 'person', Stays: 'event' };

const bottomTabNavigator = createBottomTabNavigator(
  {
    Search: SearchStack,
    Account: AccountStack,
    Stays: StaysStack,
  },
  {
    defaultNavigationOptions: ({ navigation }) => ({
      tabBarIcon: ({ focused, horizontal, tintColor }) => {
        const iconName = icons[navigation.state.routeName];

        return <Ionicons name={iconName} size={25} color={tintColor} />;
      },
    }),
    tabBarOptions: {
      activeTintColor: 'navy',
      inactiveTintColor: 'gray',
    },
  }
);

const AppNavigator = createStackNavigator(
  {
    Main: {
      screen: bottomTabNavigator,
    },
    SignIn: {
      screen: SignInModalWithQuery,
    },
    Join: {
      screen: JoinModalWithQuery,
    },
  },
  {
    mode: 'modal',
    headerMode: 'none',
  }
);

export default createAppContainer(AppNavigator);
