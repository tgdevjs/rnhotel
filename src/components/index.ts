export * from "./Calendar";
export * from "./date-range";
export * from "./HotelTitle";
export * from "./SignInButton";
