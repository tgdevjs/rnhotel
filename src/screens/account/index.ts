export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
