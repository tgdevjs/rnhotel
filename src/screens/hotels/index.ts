export * from './Hotels';
export * from './ConfirmReservation';
