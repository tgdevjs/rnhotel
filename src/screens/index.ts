export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
