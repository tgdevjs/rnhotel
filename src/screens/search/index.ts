export * from './Search';
export * from './SearchCalendar';
