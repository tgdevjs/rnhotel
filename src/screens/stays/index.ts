export * from './Stays';
export * from './StayItem';
export * from './StaysList';
export * from './StaysListGuest';
