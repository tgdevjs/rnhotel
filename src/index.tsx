/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React from 'react';
import { ApolloProvider } from 'react-apollo';

import { client, restoreCache } from './apollo';
import Navigation from './Navigation';

export default class App extends React.Component {
  state = { isSetupComplete: false };

  async componentDidMount() {
    await restoreCache();
    this.setState({ isSetupComplete: true });
  }

  render() {
    const { isSetupComplete } = this.state;

    if (!isSetupComplete) return null;
    return (
      <ApolloProvider client={client}>
        <Navigation />
      </ApolloProvider>
    );
  }
}
