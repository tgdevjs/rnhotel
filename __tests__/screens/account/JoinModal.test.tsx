import React from "react";
import { JoinModal } from "../../../src/screens/";

function joinModal(props = {}) {
  const defaultProps = {
    navigation: { goBack: jest.fn() },
    ...props
  };
  return shallow(<JoinModal {...defaultProps} />);
}

describe("JoinModal", () => {
  it("renders default", () => {
    expect(joinModal()).toMatchSnapshot();
  });
});
