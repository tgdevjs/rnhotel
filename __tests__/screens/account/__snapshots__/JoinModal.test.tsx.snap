// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinModal renders default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="name"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "margin": 10,
        "width": 200,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Button
    onPress={[Function]}
    title="Join"
  />
  <Button
    onPress={[Function]}
    title="Cancel"
  />
</View>
`;
